skepticoin/networking/peer.py

```diff
@@ -8,82 +10,62 @@
 from datetime import datetime
 from io import BytesIO
 from ipaddress import IPv6Address
-from threading import Lock
+from threading import Lock, LockType  # type: ignore
 from time import time
-from typing import Dict, List, Optional, Tuple
+from typing import Dict, List, Optional, Set, Tuple
 
 from skepticoin.__version__ import __version__
 from skepticoin.coinstate import CoinState
 from skepticoin.consensus import (
     ValidateTransactionError,
     validate_no_duplicate_output_references_in_transactions,
     validate_non_coinbase_transaction_by_itself,
-    validate_non_coinbase_transaction_in_coinstate,
-)
-from skepticoin.datatypes import Transaction
+    validate_non_coinbase_transaction_in_coinstate)
+from skepticoin.datatypes import Block, Transaction
 from skepticoin.humans import human
 from skepticoin.params import DESIRED_BLOCK_TIMESPAN
 from skepticoin.utils import block_filename, calc_work
 
-from .messages import (
-    DATA_BLOCK,
-    DATA_TRANSACTION,
-    DataMessage,
-    GetBlocksMessage,
-    GetDataMessage,
-    GetPeersMessage,
-    HelloMessage,
-    InventoryItem,
-    InventoryMessage,
-    Message,
-    MessageHeader,
-    Peer,
-    PeersMessage,
-    SupportedVersion,
-)
-from .params import (
-    EMPTY_INVENTORY_BACKOFF,
-    GET_BLOCKS_INVENTORY_SIZE,
-    GET_PEERS_INTERVAL,
-    IBD_PEER_TIMEOUT,
-    MAX_IBD_PEERS,
-    MAX_MESSAGE_SIZE,
-    PORT,
-    SWITCH_TO_ACTIVE_MODE_TIMEOUT,
-    TIME_BETWEEN_CONNECTION_ATTEMPTS,
-)
+from .messages import (DATA_BLOCK, DATA_TRANSACTION, DataMessage,
+                       GetBlocksMessage, GetDataMessage, GetPeersMessage,
+                       HelloMessage, InventoryItem, InventoryMessage, Message,
+                       MessageHeader, Peer, PeersMessage, SupportedVersion)
+from .params import (EMPTY_INVENTORY_BACKOFF, GET_BLOCKS_INVENTORY_SIZE,
+                     GET_PEERS_INTERVAL, IBD_PEER_TIMEOUT, MAX_IBD_PEERS,
+                     MAX_MESSAGE_SIZE, PORT, SWITCH_TO_ACTIVE_MODE_TIMEOUT,
+                     TIME_BETWEEN_CONNECTION_ATTEMPTS)
 
 LISTENING_SOCKET = "LISTENING_SOCKET"
-IRRELEVANT = "IRRELEVANT"
+IRRELEVANT = -1
```

Comment:
I understand why you do this, but please don't, as I don't feel like evaluating whether it will have real consequences and it's not worth it. Something I can think of off the top of my head: this string actually ends up in `peers.json`